src/core/net/ip6.cpp

```diff
@@ -984,10 +984,12 @@ otError Ip6::ProcessReceiveCallback(Message &          aMessage,
 
     if (mIsReceiveIp6FilterEnabled)
     {
+#if !OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
```

Comment:
```suggestion
#if !OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
```

Reply:
Considering our current OTBR solutions. 

| OTBR Solutions                   | PLAT_NETIF | PLAT_UDP | Should Pass RLOC/ALOC ? |
|----------------------------------|------------|----------|-------------------------|
| OT-Daemon + RCP-Dongle           | Yes        | Yes      | Yes                     |
| Wpantund + NCP-Dongle            | No         | No       | No                      |
| Wpantund + NCP-Host + RCP-Dongle | No         | Yes      | TMF Yes, Other No       |

Do you agree on this table? @bukepo @abtink @jwhui @librasungirl 

If yes, I think we should move the `RLOC/ALOC filtering code` down to be below `TMF filtering` so that we can pass TMF message but not other messages with RLOC/ALOC destinations when PLAT_UDP is ON.

Reply:
Using `!OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE` should be OK if we want to pass RLOC/ALOC dst messages when PLAT_NETIF is ON.